### PR TITLE
Changed CWL messages to JSON format.

### PR DESCRIPTION
**Pull request by Bento007**
### Examples 1
Info log with on request ID.

#### Before
```
[INFO]	2018-06-14T23:58:57.980Z	be749d7c-702e-11e8-804e-fb07b9c96f2e	Found credentials in environment variables.
```

#### After
```
{
    "levelname": "INFO",
    "asctime": "2018-06-18T16:28:35.220Z",
    "thread": 140456116651840,
    "message": "Found credentials in environment variables."
}
```

### Example 2
Info log with request ID

#### Before
```
[INFO]	2018-06-14T23:58:57.946Z	be74eb14-702e-11e8-9a7d-f175b4ca8703	[dispatch] "POST /v1/search" 206 {'replica': 'aws'}
```

#### After
```
{
    "levelname": "INFO",
    "asctime": "2018-06-14T23:58:57.946Z",
    "aws_request_id": "d966be4a-7305-11e8-a3ca-ab03f081e825",
    "thread": 140249860183808,
    "message": "[dispatch] "POST /v1/search" 206 {'replica': 'aws'}"
}
```

### CODE REVIEW

File: dss/api/search.py [hunk 1]
```diff
@@ -57,28 +57,28 @@ def post(json_request_body: dict,
         return response
     except TransportError as ex:
         if ex.status_code == requests.codes.bad_request:
-            logger.debug(f"Invalid Query Recieved. Exception: {ex}")
+            logger.debug("Invalid Query Recieved.", exc_info=ex, stack_info=True)
```
Comment (hannes-ucsc) [inline dss/api/search.py]: received

File: dss/logging.py [hunk 2]
```diff
@@ -42,25 +46,73 @@
...
+
+
 def configure_cli_logging():
     """
     Prepare logging for use in a command line application.
     """
-    _configure_logging(stream=sys.stderr)
+    logHandler = logging.StreamHandler(stream=sys.stderr)
+    formatter = DSSJsonFormatter()
+    logHandler.setFormatter(formatter)
```
Comment (hannes-ucsc) [inline dss/logging.py]: Any way we can avoid the repetition?

Comment (Bento007) [reply dss/logging.py]: will fix

File: dss/api/search.py (hunk 1 above)
Comment (Bento007) [reply dss/api/search.py]: will fix

File: dss/logging.py [hunk 3]
```diff
@@ -42,25 +46,97 @@
...
+        log_record.update(message_dict)
+        merge_record_extra(record, log_record, reserved=self._skip_fields)
+
+
+class DispatchFilter(logging.Filter):
+    def filter(self, record):
+        return False if '[dispatch]' in record.msg else True
+
+
+def get_json_log_handler():
```
Comment (hannes-ucsc) [inline dss/logging.py]: Suggested: Consider making this protected.

File: dss/logging.py [hunk 4]
```diff
@@ -42,25 +46,97 @@
...
+    log_handler.setFormatter(DSSJsonFormatter())
+    return log_handler
+
+
 def configure_cli_logging():
     """
     Prepare logging for use in a command line application.
     """
-    _configure_logging(stream=sys.stderr)
+    log_handler = get_json_log_handler()
```
Comment (hannes-ucsc) [inline dss/logging.py]: Suggested: Consider inlining this.

File: dss/logging.py (hunk 3 above)
Comment (Bento007) [reply dss/logging.py]: I'll make it protected

File: dss/logging.py (hunk 4 above)
Comment (Bento007) [reply dss/logging.py]: Inlined it.

**Comment by Bento007** [comment]
@hannes-ucsc 

> Do you have an example of what the JSON-formatted log messages look like in ELK?

Not yet.

> Why aren't all logger invocations making use of the extras= mechanism?

I didn't want to modify all of the log messages in this PR. We should modify them as needed

> What's your reasoning for adding stack_info=True?

I didn't know there would be that much overlap. I remove it when `exc_info` is already included.

**Comment by xbrianh** [comment]
What's the status of this?

**Comment by Bento007** [comment]
it's probably not going to happen

**Pull request closed without merging**